### ISSUE TITLE
Warn about useless imports

### ISSUE THREAD

**Issue by CryZe** [rust-lang/rust#68177]
I'm seeing quite a lot of people do the following:
```rust
use some_crate;
```
However due to uniform paths, this doesn't do anything, as the crate is already accessible like that in the current module. This neither disambiguates nor does it have any meaningful effect, yet Rust does not emit a warning.

All of this only applies to uniform paths. So in Rust 2015 this actually is a meaningful import.

**Comment by mati865** [comment]
Can you provide better example?
Works as expected for:
```rust
use rand;

pub fn foo() {}
```
Warning:
```
warning: unused import: `rand`
 --> src/lib.rs:1:5
  |
1 | use rand;
  |     ^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

**Comment by CryZe** [comment]
This is the warning about rand not being used at all. You actually need to use rand.

Here's an example with std on the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3bef949341cea0bff47a4c3cddd49071)
```rust
use std;

fn main() {
    std::iter::once(5);
}
```

The `use std;` there doesn't do anything, as `std::iter::once(5)` works regardless, as std is already accessible in the module. Yet there is no warning.

**Comment by CryZe** [comment]
Here is where I most recently encountered this in the wild: https://github.com/wasmerio/wasmer/pull/1140

There they have code that roughly looks like this:
```rust
use blake3;

fn main() {
    let mut hasher = blake3::Hasher::new();
    ...
}
```
They `use blake3;` despite `blake3::Hasher::new()` working just fine without the `use blake3;`.

I've seen people make this mistake quite often by now, as this used to be required in Rust 2015 but is now completely redundant in Rust 2018.

**Comment by petrochenkov** [comment]
This is a duplicate of https://github.com/rust-lang/rust/issues/61640 and https://github.com/rust-lang/rust/issues/10178 (which contains some implementation hints), and the lint was partially implemented in https://github.com/rust-lang/rust/pull/58805.
